xbmc/XBDateTime.cpp

```diff
@@ -1572,3 +1575,41 @@ int CDateTime::MonthStringToMonthNum(const std::string& month)
 
   return i;
 }
+std::string CDateTime::GetISOorLocalizedDate(std::string& strIsoDate) const
```

Comment:
XBDateTime unit now depends on ServiceBroker and settings units, and that doesn't seem right,  datetime should be lower level than that.  Despite the urge to simplify all GUI getting string calls I don't think that the decision of what shows should be made in `CDateTime`.

This routine is also just a string -> string (string reformatting) utility, not really a `CDateTime` method per se, and not a property "getter" method which it itinitally could be mistaken for.

My suggestion is move the processing to convert (partial) ISO date strings to (partial) localized date stings to a `ISODateToLocalizedDate()` routine in StringUtils, next to  'StringUtils::DateStringToYYYYMMDD()`.

Checking the setting before calling that string formatting in the 3 places that use is OK, at least they are all in the GUI layer which is where "how for format and show" something decisons belong.

Reply:
You could counter it is a datetime utility not just a string one, and date stuff is best together but it isn't a CDateTime utility and we already have date stuff in StringUtils.

Reply:
OK.  Consider it moved.